android/app/src/main/java/com/zulipmobile/notifications/FcmListenerService.kt

```diff
@@ -0,0 +1,22 @@
+package com.zulipmobile.notifications
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.zulipmobile.MainApplication
+import com.facebook.react.ReactApplication
+
+class FcmListenerService : FirebaseMessagingService() {
+    override fun onMessageReceived(message: RemoteMessage) {
+        if (applicationContext !is MainApplication) return
```

Comment:
This logic is a bit puzzling; it made me wonder why we have it. The one line of the function after this doesn't have an obvious reason to care about the type of the `applicationContext`.

It looks like the answer is that this function used to interact with `applicationContext`, until eaa8b4009bbb62429da6559d07c4507fe0efc9f1 . So until then, this was needed in order for that to type-check.

Now we can just take it out.